server/channels/store/sqlstore/group_store.go

```diff
@@ -1284,34 +1284,42 @@ func (s *SqlGroupStore) ChannelMembersToRemove(channelID *string) ([]*model.Chan
 }
 
 func (s *SqlGroupStore) groupsBySyncableBaseQuery(st model.GroupSyncableType, t selectType, syncableID string, opts model.GroupSearchOpts) sq.SelectBuilder {
-	selectStrs := map[selectType]string{
-		selectGroups:      "UserGroups.*, gs.SchemeAdmin AS SyncableSchemeAdmin",
-		selectCountGroups: "COUNT(*)",
+	var query sq.SelectBuilder
+	switch t {
+	case selectGroups:
+		query = s.userGroupsSelectQuery.
+			Column("gs.SchemeAdmin AS SyncableSchemeAdmin")
+	case selectCountGroups:
+		query = s.getQueryBuilder().
+			Select("COUNT(*)").
+			From("UserGroups")
 	}
 
-	var table string
-	var idCol string
 	if st == model.GroupSyncableTypeTeam {
-		table = "GroupTeams"
-		idCol = "TeamId"
+		query = query.
+			Join("GroupTeams gs ON gs.GroupId = UserGroups.Id").
+			Where(sq.Eq{
+				"gs.TeamId":   syncableID,
+				"gs.DeleteAt": 0,
+			})
 	} else {
-		table = "GroupChannels"
-		idCol = "ChannelId"
+		query = query.
+			Join("GroupChannels gs ON gs.GroupId = UserGroups.Id").
+			Where(sq.Eq{
+				"gs.ChannelId": syncableID,
+				"gs.DeleteAt":  0,
+			})
 	}
 
-	query := s.getQueryBuilder().
-		Select(selectStrs[t]).
-		From(fmt.Sprintf("%s gs", table)).
-		LeftJoin("UserGroups ON gs.GroupId = UserGroups.Id").
-		Where(fmt.Sprintf("UserGroups.DeleteAt = 0 AND gs.%s = ? AND gs.DeleteAt = 0", idCol), syncableID)
+	query = query.
+		Where(sq.Eq{
+			"UserGroups.DeleteAt": 0,
```

Comment:
Technically, we could lift this up into the start of the method where we initialized the query, since it's common to both. I opted to keep it here to make the review clearer, but would happily simplify.